executor/src/witgen/data_structures/finalizable_data.rs

```diff
@@ -36,10 +50,11 @@ pub struct CompactData<T> {
 
 impl<T: FieldElement> CompactData<T> {
     /// Creates a new empty compact data storage.
-    pub fn new(column_ids: &[PolyID]) -> Self {
-        let col_id_range = column_ids.iter().map(|id| id.id).minmax();
-        let (first_column_id, last_column_id) = col_id_range.into_option().unwrap();
-        let column_count = (last_column_id - first_column_id + 1) as usize;
+    pub fn new<'a>(column_ids: impl Iterator<Item = &'a PolyID>) -> Self {
+        let ColumnLayout {
+            column_count,
+            first_column_id,
+        } = ColumnLayout::from_id_list(column_ids);
```

Comment:
Shouldn't we just store the layout in `CompactData`, instead of the fields?

Reply:
I thought about this as well, but it's so many changes...

Reply:
Ok, I'll do it...